dapps/marketplace/src/components/DesktopUserActivation.js

```diff
@@ -70,9 +71,28 @@ class UserActivation extends Component {
       code: '',
       firstNameError: null
     }
+
+    this.EnrollButton = WithEnrolmentModal('button')
+
+    if (state.stage !== props.stage) {
+      this.onStageChanged()
+    }
+  }
+
+  componentDidMount() {
+    this.mounted = true
```

Comment:
this isn't ideal... can we avoid doing things this way?

Reply:
The thing is the modals have an exit animation, so there is a timeout after which the `onClose` prop will be called. By the time this prop gets called for the last modal, the component would have already unmounted. 

Just read that [it is antipattern](https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html). Will change it. Thanks for the heads up.